### DISCUSSION TITLE
WNS returns -0.62 while TNS returns 0

### DISCUSSION THREAD

**Discussion by xiaosiCU** [The-OpenROAD-Project/OpenROAD-flow-scripts#1917] [Q&A]
I have successfully run the complete flow for the Black Parrot design using the following command:

```
make final -DESIGN_CONFIG=OpenROAD-flow-scripts/flow/designs/nangate45/black_parrot/config.mk
```

After executing the complete flow, I am interested in examining the changes in timing metrics at different stages of the design process. To that end, I have executed placement with:

```
make do-place -DESIGN_CONFIG=OpenROAD-flow-scripts/flow/designs/nangate45/black_parrot/config.mk
```

For reporting timing metrics, I appended both global_place.tcl and resize.tcl scripts with the following commands:

``` 
report_wns
report_tns
```
As a result, I obtained the following metrics:
After global placement:

```
WNS: -241.73
TNS: -1048567.38
```

And after resizing:

```
WNS: -0.62
TNS: 0
```

Given that there is a negative slack, I am puzzled as to why the TNS would return a result of 0. Could this be an issue of precision rounding?

Furthermore, if I aim to optimize the existing macro placement and global placement algorithms specifically for timing metrics, at which step in the entire OpenROAD process do you believe it would be most valuable to consider report_timing? For instance, should I look at it post-CTS, post-detailed routing, or after resizing? Reporting early might show very poor timing metrics; however, after CTS or detailed routing, the timing may meet the requirements, making it difficult to discern differences between algorithms.

Thank you very much for your time and assistance!

**Comment by maliberty** [comment]
That seems like a bug.  Is it possible for you to package a test case and open an issue?

**Reply by xiaosiCU** [reply]
Yes of course. But I can hardly find an instruction for packing a test case. Any tutorials?

**Comment by maliberty** [comment]
If you are using ORFS then the issue creation page gives some guidance
```
A. Use make <SCRIPT_NAME>_issue to create a tar file with all the files to reproduce the bug(s).

    Steps:

    1. Head to the flow directory e.g "./flow"

    2. Set the ISSUE_TAG variable to rename the generated tar file

    3. Run make {script}_issue in the directory "OpenROAD-flow-scripts/flow" where script is wildcarded from the "OpenROAD-flow-scripts/scripts" directory e.g make cts_issue

    4. Upload the generated tar file
 ```
